assets/js/googlesitekit/datastore/user/user-info.test.js

```diff
@@ -364,5 +364,53 @@ describe( 'core/user userInfo', () => {
 				expect( global[ userDataGlobal ] ).not.toEqual( undefined );
 			} );
 		} );
+
+		describe( 'getAccountChooserURL', () => {
+			it( 'throws an error if a destinationURL is not given', () => {
+				expect( () => {
+					registry.select( CORE_USER ).getAccountChooserURL();
+				} ).toThrow( 'destinationURL is required' );
+			} );
+
+			it( 'returns the encoded destination url with the email appended', async () => {
+				global[ userDataGlobal ] = userData;
+
+				const testURL = 'https://analytics.google.com/dashboard/';
+				const userEmail = global[ userDataGlobal ].user.email;
+
+				registry.select( CORE_USER ).getAccountChooserURL( testURL );
+
+				await subscribeUntil(
+					registry,
+					() =>
+						registry
+							.select( CORE_USER )
+							.getAccountChooserURL( testURL ) !== undefined
+				);
+
+				const accountChooserURL = registry
+					.select( CORE_USER )
+					.getAccountChooserURL( testURL );
+
+				expect( accountChooserURL ).toBe(
+					`https://accounts.google.com/accountchooser?continue=${ encodeURIComponent(
+						testURL
+					) }&Email=${ encodeURIComponent( userEmail ) }`
+				);
```

Comment:
I feel like this test would be a bit nicer to read if the end-result text was here as a string. Maybe we could use an inline snapshot test here instead, so we aren't just replicating the code here that's in the selector itself, eg:

```
				expect( accountChooserURL ).toMatchInlineSnapshot();
```